internal/ingest/linksystem.go

```diff
@@ -289,9 +297,23 @@ func (ing *Ingester) storageHook(pubID peer.ID, c cid.Cid) {
 }
 
 // syncAdEntries fetches all the entries for a single advertisement
-func (ing *Ingester) syncAdEntries(from peer.ID, ad schema.Advertisement, adCid cid.Cid, prevWait <-chan struct{}, unlock context.CancelFunc) {
-	<-prevWait
-	defer unlock()
+func (ing *Ingester) syncAdEntries(from peer.ID, ad schema.Advertisement, adCid, prevCid cid.Cid) {
+	if !ing.adWaiter.wait(prevCid) {
+		// If there was a previous ad, but it was not waitiable, then check if
```

Comment:
I don't love this. Is there some other solution we can come up with that is better?

Reply:
I think I can do better by building a linked list of ads as they arrive.  The indexer can then wait until all the ads in the chain are received (when the initial ad with no previous link is received), or until the chain links to a previously processed ad.

Reply:
Am I right to say without this PR the index processing is broken on the indexer node? If so, It would be great to push this out sooner rather than later if you folks think the implementation is correct and improve later.

Reply:
@masih Yes, it is effectively broken without this.  Will merge and we can provide improvements in following PR